src/main/java/CLI/Main.java

```diff
@@ -375,4 +387,91 @@ private static void checkHistoryAddJobsAndDownload(Job job, boolean removeInputB
+                Link = "https://github.com/SaptarshiSarkar12/Drifty/releases/latest/download/Drifty-CLI_linux";
+                String fileName = "Drifty-CLI_linux";
+                String dirPath = ".drifty/updates";
+                FileDownloader downloader = new FileDownloader(Link, fileName, dirPath);
+                downloader.run();
+                newFilePath = dirPath +'\\' + fileName;
+            }
+            Updater.replaceUpdate(oldFilePath , newFilePath);
+            return true;
+        }
+        return false;
+    }
+
+    private static String getLatestVersion() {
+        try {
+            URL url = new URI("https://api.github.com/repos/SaptarshiSarkar12/Drifty/releases/latest").toURL();
+            HttpURLConnection connection = (HttpURLConnection) url.openConnection();
+            connection.setRequestMethod("GET");
+
+            BufferedReader reader = new BufferedReader(new InputStreamReader(connection.getInputStream()));
+            String response = reader.readLine();
+            reader.close();
+
+            // Parse JSON response to get the "tag_name"
+            // For simplicity, we're assuming the tag_name is a string enclosed in double quotes
+            int start = response.indexOf("\"tag_name\":\"") + 12;
+            int end = response.indexOf("\"", start);
+            System.out.println( response.substring(start, end));
+            return response.substring(start, end);
+        } catch (Exception e) {
+            e.printStackTrace();
+            return null;
+        }
+    }
+
+    private static boolean isNewerVersion(String newVersion, String currentVersion) {
+        // Split version strings into arrays of integers
+        String[] newVersionParts = newVersion.replaceAll("[^\\d.]", "").split("\\.");
+        String[] currentVersionParts = currentVersion.replaceAll("[^\\d.]", "").split("\\.");
+
+        // Convert parts to integers
+        int[] newVersionNumbers = new int[newVersionParts.length];
+        int[] currentVersionNumbers = new int[currentVersionParts.length];
+
+        for (int i = 0; i < newVersionParts.length; i++) {
+            newVersionNumbers[i] = Integer.parseInt(newVersionParts[i]);
+        }
+
+        for (int i = 0; i < currentVersionParts.length; i++) {
+            currentVersionNumbers[i] = Integer.parseInt(currentVersionParts[i]);
+        }
+
+        // Compare version numbers
+        for (int i = 0; i < Math.min(newVersionNumbers.length, currentVersionNumbers.length); i++) {
+            if (newVersionNumbers[i] > currentVersionNumbers[i]) {
+                return true; // New version is greater
+            } else if (newVersionNumbers[i] < currentVersionNumbers[i]) {
+                return false; // Current version is greater
+            }
+        }
+
+        // If all compared parts are equal, consider the longer version as newer
+        return newVersionNumbers.length > currentVersionNumbers.length;
+    }
+}
```

Comment:
@kmr-rohit Please consider optimizing it.
This check can be optimized similar to a palindrome program.

---
I recommend this way:
- Check the first number of the version numbers before the first dot (`.`). If they are different, then update is available else proceed below.
    > For example, if old version is `1.3.0` and new version is `2.0.0`, then, as 2 and 1 are different, then, update is present.
- Similarly, check for the next digits in the version number.
Time Complexity will reduce, which will make the program much faster comparatively.

Reply:
Implemented logic shared, removed unnecessary redundant code.

Reply:
Okay :+1:. Thank you!